src/glance.gleam

```diff
@@ -1299,10 +1306,15 @@ fn record_update_field(
   tokens: Tokens,
 ) -> Result(#(#(String, Expression), Tokens), Error) {
   case tokens {
-    [#(t.Name(name), _), #(t.Colon, _), ..tokens] -> {
-      use #(expression, tokens) <- result.try(expression(tokens))
-      Ok(#(#(name, expression), tokens))
-    }
+    [#(t.Name(name), _), #(t.Colon, _), ..tokens] ->
+      case tokens {
+        [#(t.Comma, _), ..] | [#(t.RightParen, _), ..] ->
+          Ok(#(#(name, Variable(name)), tokens))
+        _ -> {
+          use #(expression, tokens) <- result.try(expression(tokens))
+          Ok(#(#(name, expression), tokens))
```

Comment:
Here I'd change the underlying data structure to make sure one can tell the difference between a punned field or not. Maybe defining a `RecordUpdateField`

Reply:
I was wondering about that as well. I have a feeling the regular `Field` type should be fine here, and the code here perhaps wasn't updated to use it.

I'll try switching it to `Field` and see if that works!

Reply:
Ah, I think I might see why `Field` wasn't used. A `Field` has an *optional* label, but labels are *required* in a record update.

That doesn't cause any parsing issues I don't believe, but it would introduce a bit of unnecessary pattern matching/`let assert`ing on the consumer side since you can guarantee a successfully parsed record update WILL have a `Some` label.

Reply:
Perhaps the consistency is worth the slight inconvenience?

Reply:
No, I don't think it is worth it. I'd make sure that the impossible case is not representable by adding another ad hoc type

Reply:
Sounds good to me!

Reply:
This would be resolved by proposal in the main comment thread: https://github.com/lpil/glance/pull/20#issuecomment-2307480917